pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -175,10 +189,12 @@ private void processExistingSegment(SegmentMetadata segmentMetadata, URI finalSe
         }
 
         _pinotHelixResourceManager
-            .refreshSegment(tableNameWithType, segmentMetadata, existingSegmentZKMetadata, zkDownloadURI, crypter);
+            .refreshSegment(tableNameWithType, segmentMetadata, existingSegmentZKMetadata,
+                replaceOnly ? expectedVersion : -1, zkDownloadURI, crypter);
```

Comment:
It seems this code `replaceOnly ? expectedVersion : -1` appears in multiple places. Is it possible to do only one check and then pass this final value to the target places?

Reply:
+1. We can compute `expectedVersion` once and reuse the same value

Reply:
Same here, always check expected version regardless of the flag